client/src/components/Calendar.jsx

```diff
@@ -16,6 +16,65 @@ class Calendar extends React.Component {
     };
   }
 
+  componentDidMount() {
+    this.props.getCalendarData((err, responseData) => {
+      if (err) {
+        console.log('Response Error: ', err);
+      } else {
+        var listingData = responseData.data.listing;
+        var bookingData = responseData.data.bookings;
+
+        //sort booking data by date in ascending order 
+        bookingData.sort(function(a, b) {
+          return new Date(a.checkin) - new Date(b.checkin);
+        });
+
+        this.setState({
+          listing: listingData,
+          bookings: bookingData
+        });
+      }
+    });
+  }
+
+  onDayClick(e) {
```

Comment:
this is really helpful for me to see... i've been trying to think of an approach to figure out this same problem for myself. cool to see you've gotten it figured out.

Reply:
Glad this helped you!